pkg/cmd/cli/cmd/volume.go

```diff
@@ -0,0 +1,596 @@
+			if err != nil {
+				errList = append(errList, err)
+			}
+			if len(source) > 1 {
+				errList = append(errList, errors.New("must provide only one volume for --source"))
+			}
+
+			var vs kapi.VolumeSource
+			err = json.Unmarshal([]byte(a.Source), &vs)
+			if err != nil {
+				errList = append(errList, err)
+			}
+		}
+	} else if len(a.Source) > 0 || len(a.Path) > 0 || len(a.SecretName) > 0 || len(a.ClaimName) > 0 {
+		errList = append(errList, errors.New("--type|--path|--secret-name|--claim-name|--source are only valid for --add operation"))
+	}
+
+	return kerrors.NewAggregate(errList)
+}
+
+func (v *VolumeOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer) error {
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+	v.OutputVersion = kcmdutil.OutputVersion(cmd, clientConfig.Version)
+
+	cmdNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+
+	v.DefaultNamespace = cmdNamespace
+	v.Writer = out
+	v.Mapper = mapper
+	v.Typer = typer
+	v.RESTClientFactory = f.Factory.RESTClient
+	v.UpdatePodSpecForObject = f.UpdatePodSpecForObject
+
+	if v.Add && len(v.Name) == 0 {
+		v.Name = string(util.NewUUID())
+	}
+	// In case of volume source ignore the default volume type
+	if len(v.AddOpts.Source) > 0 {
+		v.AddOpts.Type = ""
+	}
+	return nil
+}
+
+func (v *VolumeOptions) RunVolume(args []string) error {
+	b := resource.NewBuilder(v.Mapper, v.Typer, resource.ClientMapperFunc(v.RESTClientFactory)).
+		ContinueOnError().
+		NamespaceParam(v.DefaultNamespace).DefaultNamespace().
+		SelectorParam(v.Selector).
+		ResourceTypeOrNameArgs(v.All, args...).
+		Flatten()
+
+	one := false
+	infos, err := b.Do().IntoSingular(&one).Infos()
+	if err != nil {
+		return err
+	}
+
+	skipped := 0
+	for _, info := range infos {
+		ok, err := v.UpdatePodSpecForObject(info.Object, func(spec *kapi.PodSpec) error {
+			var e error
+			if v.Add {
+				e = v.addVolumeToSpec(spec)
+			} else if v.Remove {
+				e = v.removeVolumeFromSpec(spec)
+			} else if v.List {
+				e = v.listVolumeForSpec(spec, info)
+			}
+			return e
+		})
+		if !ok {
+			skipped++
+			continue
+		}
+		if err != nil {
+			fmt.Fprintf(v.Writer, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, err)
+			continue
+		}
+	}
+	if one && skipped == len(infos) {
+		return fmt.Errorf("the %s %s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+	}
+
+	if v.List {
+		return nil
+	}
+
+	objects, err := resource.AsVersionedObject(infos, false, v.OutputVersion)
+	if err != nil {
+		return err
+	}
+
+	if len(v.Output) != 0 {
+		p, _, err := kubectl.GetPrinter(v.Output, "")
+		if err != nil {
+			return err
+		}
+		return p.PrintObj(objects, v.Writer)
+	}
+
+	failed := false
+	for _, info := range infos {
+		data, err := info.Mapping.Codec.Encode(info.Object)
+		if err != nil {
+			fmt.Fprintf(v.Writer, "Error: %v\n", err)
```

Comment:
Instead of printing the error here and using the `failed` control, you could use an `err` array and `errors.NewAggregate(errs)`.